src/coreclr/vm/genericdict.cpp

```diff
@@ -992,7 +995,7 @@ Dictionary::PopulateEntry(
                 if (nonExpansive)
                     return NULL;
 
-                pOwnerMT = ownerType.GetMethodTable();
+                pOwnerMT = ownerType.IsByRef() ? ownerType.GetTypeParam().GetMethodTable() : ownerType.GetMethodTable();
```

Comment:
This type is expected to be a type with methods. We should not ever end up with byref here.

Reply:
It should not be a problem once the other feedback is addressed.